File: intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, bool> mp;
        for(int i : nums1)
            mp[i] = true;

        vector<int> ans;
        for(int i : nums2)
            if(mp[i])
                mp[i] = false, ans.push_back(i);

        return ans;
    }
};

File: intersection-of-two-arrays.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        Dictionary<int, bool> mp = new();
        foreach(int i in nums1)
            mp[i] = true;

        List<int> ans = new();
        foreach(int i in nums2){
            if(mp.ContainsKey(i)){
                mp.Remove(i);
                ans.Add(i);
            }
        }

        return ans.ToArray();
    }
}